ComponentKit/StatefulViews/CKStatefulViewReusePool.mm

```diff
@@ -10,40 +10,76 @@
 #import "CKStatefulViewComponentController.h"
 
 #import "CKStatefulViewReusePool.h"
+#import "CKAssert.h"
 
 #import <unordered_map>
 
+struct FBStatefulReusePoolItemEntry {
+  UIView *view;
+  CKStatefulViewReusePoolPendingMayRelinquishBlock block;
+};
+
 class FBStatefulReusePoolItem {
```

Comment:
I decided to just reuse this custom multimap for my new pending pool and just add on the block API. For the normal pool, the block will be nil.